src/components/Search/SearchForm.jsx

```diff
@@ -1,53 +1,84 @@
 import React from 'react';
 import { connect } from 'react-redux';
 import { browserHistory } from 'react-router';
+import { debounce } from 'lodash';
 import TextField from 'material-ui/TextField';
-// import RaisedButton from 'material-ui/RaisedButton';
-// import ActionSearch from 'material-ui/svg-icons/action/search';
-// import CircularProgress from 'material-ui/CircularProgress';
 import { getSearchResultAndPros, setSearchQuery } from 'actions';
 import styles from './search.css';
 
-const SearchForm = ({ dispatchSearch, dispatchSetQuery, query }) => {
-  const formSubmit = (e) => {
+class SearchForm extends React.Component {
+  constructor(props) {
+    super(props);
+
+    let query = '';
+    const location = browserHistory.getCurrentLocation();
+    if (location.pathname === '/search') {
+      // Avoid getting ?q= from other paths
+      query = location.query.q;
+    }
+    this.state = { query };
```

Comment:
FYI, I couldn't get it from the store because it seems like it updates after the search component mounts. Doing it in componentWillReceiveProps might cause some race conditions.